frontend/ui/widget/screenshoter.lua

```diff
@@ -37,59 +35,74 @@ function Screenshoter:init()
+        keep_dialog_open = true,
+        cancel_text = _("Close"),
+        cancel_callback = function()
+            if when_done_func then when_done_func() end
+        end,
+        ok_text = _("Set as screensaver"),
         ok_callback = function()
             G_reader_settings:saveSetting("screensaver_type", "image_file")
             G_reader_settings:saveSetting("screensaver_image", screenshot_name)
+            UIManager:close(confirm_box)
+            if when_done_func then when_done_func() end
         end,
-        cancel_text = _("No"),
+        other_buttons_first = true,
+        other_buttons = {{
+            {
+                text = _("Delete"),
+                callback = function()
+                    local __ = os.remove(screenshot_name)
+                    UIManager:close(confirm_box)
+                    if when_done_func then when_done_func() end
+                end,
+            },
+            {
+                text = _("View"),
+                callback = function()
+                    local image_viewer = require("ui/widget/imageviewer"):new{
+                        file = screenshot_name,
+                        modal = true,
+                        with_title_bar = false,
+                        buttons_visible = true,
+                    }
+                    UIManager:show(image_viewer)
+                end,
+            },
+        }},
     }
-    UIManager:show(widget)
+    UIManager:show(confirm_box)
     -- trigger full refresh
     UIManager:setDirty(nil, "full")
     return true
 end
 
 function Screenshoter:chooseFolder()
-    local buttons = {}
-    table.insert(buttons, {
-        {
-            text = _("Choose screenshot folder"),
-            callback = function()
-                UIManager:close(self.choose_dialog)
-                require("ui/downloadmgr"):new{
-                    onConfirm = function(path)
-                        G_reader_settings:saveSetting("screenshot_dir", path .. "/")
-                        UIManager:show(InfoMessage:new{
-                            text = T(_("Screenshot folder set to:\n%1"), BD.dirpath(path)),
-                            timeout = 3,
-                        })
-                    end,
-                }:chooseDir()
-            end,
-        }
-    })
-    table.insert(buttons, {
-        {
-            text = _("Close"),
-            callback = function()
-                UIManager:close(self.choose_dialog)
-            end,
-        }
-    })
     local screenshot_dir = G_reader_settings:readSetting("screenshot_dir") or DataStorage:getDataDir() .. "/screenshots/"
-    self.choose_dialog = ButtonDialogTitle:new{
-        title = T(_("Current screenshot folder:\n%1"), BD.dirpath(screenshot_dir)),
-        buttons = buttons
+    local confirm_box = ConfirmBox:new{
+        text = T(_("Screenshot folder is set to:\n%1\n\nDo you want to choose new folder for screenshots?"), screenshot_dir),
```

Comment:
With do you want should be fine too, but definitely *a* folder.
```suggestion
        text = T(_("Screenshot folder is set to:\n%1\n\nChoose a new folder for screenshots?"), screenshot_dir),
```